components/hab/src/command/pkg/env.rs

```diff
@@ -12,18 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::HashMap;
 use std::path::Path;
 
 use hcore::package::{PackageIdent, PackageInstall};
 
 use error::Result;
 
-// TODO: This needs a windows compatible version
 pub fn start(ident: &PackageIdent, fs_root_path: &Path) -> Result<()> {
     let pkg_install = PackageInstall::load(ident, Some(fs_root_path))?;
     let env = pkg_install.environment_for_command()?;
+    render_environemt(env)?;
+    Ok(())
+}
+
+#[cfg(not(windows))]
+fn render_environemt(env: HashMap<String, String>) -> Result<()> {
     for (key, value) in env.into_iter() {
         println!("export {}=\"{}\"", key, value);
     }
     Ok(())
 }
+
+#[cfg(windows)]
+fn render_environemt(env: HashMap<String, String>) -> Result<()> {
+    use hcore::fs::FS_ROOT_PATH;
+    use std::env;
+    use std::path::PathBuf;
+
+    let mut paths: Vec<PathBuf> = match env.get("PATH") {
+        Some(path) => env::split_paths(&path).collect(),
+        None => vec![],
+    };
+    for i in 0..paths.len() {
+        if paths[i].starts_with("/") {
+            paths[i] = Path::new(&*FS_ROOT_PATH).join(paths[i].strip_prefix("/").unwrap());
+        }
+    }
```

Comment:
Does this logic exist anywhere else in the code? It seems like something that would be generally useful as some kind of library function.

Reply:
It does. I just added https://github.com/habitat-sh/core/pull/86 to move this prepending there.